models.py

```diff
@@ -363,23 +352,22 @@ def _compute_emissions(self,
 
         # apply Bidirectional LSTM
         if self.uses_lstm:
-            seq_lengths = torch.sum(mask.int(),dim = 1)
+            if mask is None:
+                # shape (batch_size, seq_length)
+                mask = self._get_mask_for(words)   
+            
+            seq_lengths = torch.sum(mask.int(), dim=1)
             seq_lengths, sent_perm = seq_lengths.sort(0, descending=True)
             inputs = inputs[sent_perm]
 
-            packed_input = pack_padded_sequence(inputs, seq_lengths.cpu().data.numpy(), batch_first=True)
-     
+            packed_input = pack_padded_sequence(inputs, seq_lengths.data.tolist(), batch_first=True)
```

Comment:
Can just `seq_lengths.data` work? Calling `.tolist()` is *very* slow, up to 1000x times slower than `.numpy()`. If just `.data` works then use that, no need to call `.numpy()`. If it doesn't, then `.numpy()` is preferred as it's faster.